builtins/amp-img.js

```diff
@@ -64,17 +64,12 @@ export class AmpImg extends BaseElement {
   }
 
   /** @override */
-  prerenderAllowed() {
-    return true;
-  }
-
-  /** @override */
   isRelayoutNeeded() {
     return true;
   }
 
   /** @override */
-  layoutCallback() {
+  prerenderCallback() {
```

Comment:
> 1. We need to make sure that "prerendered" elements are not marked as "laid out". In the current implementation they are.
This implementation uses a separate state for managing the prerendered state - so the element normal state won't be laid out when they are prerendered.
> 2. We need to decide whether we will always run "prerender". Obviously we will run it if we are NOT ready to layout. But we will we run it along with layout as well? It looks like so far we DO want to run it always because layout can take a lot longer.
The reason I always do prerender is because by assumption we expect prerendering to be much faster than layouting (loading a small thumbnail vs. loading a whole iframe) and running both together allows us to display something sooner than we would have otherwise (YT thumbnail will appear faster than the player itself) this is specially true on mobile and slower connections.

>Also, some approaches such as placeholder can yield better results since runtime will manage them better w/o assistance. For instance, in this PR amp-youtube is kind of regressing - currently we add placeholder in the buildCallback and it immediately becomes the responsibility of the runtime and its rules. Let's review the effect on other elements one more time to make sure we are putting the right intentions into the prerender callback.
The `amp-youtube` is not regressing really, `prerenderCallback` is called before `layoutCallback` the AMP runtime manages `placeholder` in `firstLayoutComplete` callback which by then we'll have already created and appended the `placeholder` element so AMP runtime still manages it correctly. That being said, it might still be clearer to move the DOM building for the placeholder inside `buildCallback` instead. I think that makes sense.